scripts/exrrfs_jedivar.sh

```diff
@@ -1,40 +1,75 @@
 #!/usr/bin/env bash
 declare -rx PS4='+ $(basename ${BASH_SOURCE[0]:-${FUNCNAME[0]:-"Unknown"}})[${LINENO}]${id}: '
 set -x
+
 cpreq=${cpreq:-cpreq}
-prefix='GFS'
 BUMPLOC=${BUMPLOC:-"conus12km-401km11levels"}
 
 cd ${DATA}
+
 CDATEm1=$($NDATE -1 ${CDATE})
 start_time=$(date -d "${CDATE:0:8} ${CDATE:8:2}" +%Y-%m-%d_%H:%M:%S) 
 timestr=$(date -d "${CDATE:0:8} ${CDATE:8:2}" +%Y-%m-%d_%H.%M.%S) 
+#
 # determine whether to begin new cycles
-IFS=' ' read -r -a array <<< "${PROD_BGN_AT_HRS}"
-begin="NO"
-for hr in "${array[@]}"; do
-  if [[ "${cyc}" == "$(printf '%02d' ${hr})" ]]; then
-    begin="YES"; break
-  fi
-done
+#
+if [[ -r "${UMBRELLA_ROOT}/prep_ic/init.nc" ]]; then
+  start_type='cold'
+  initial_filename='init.nc'
+  initial_file=${UMBRELLA_ROOT}/prep_ic/init.nc
+else
+  start_type='warm'
+  initial_filename='mpasin.nc'
+  initial_file=${UMBRELLA_ROOT}/prep_ic/mpasin.nc
+fi
+#
+#
 #
 ${cpreq} ${FIXrrfs}/physics/${PHYSICS_SUITE}/* .
-ln -snf VEGPARM.TBL.da VEGPARM.TBL #gge.debug temp
+#ln -snf VEGPARM.TBL.da VEGPARM.TBL #gge.debug temp
 mkdir -p graphinfo stream_list
 ${cpreq} ${FIXrrfs}/graphinfo/* graphinfo/
 ${cpreq} ${FIXrrfs}/jedi/obsop_name_map.yaml .                  
 ${cpreq} ${FIXrrfs}/jedi/keptvars.yaml .              
 ${cpreq} ${FIXrrfs}/jedi/geovars.yaml . 
 ${cpreq} ${FIXrrfs}/stream_list/${PHYSICS_SUITE}/* stream_list/
+#
+# create data directory 
+#
 mkdir -p data; cd data                   
 mkdir -p bumploc obs ens
-${cpreq} ${FIXrrfs}/bumploc/${BUMPLOC} bumploc/
+
+#
+#  bump files
+#
+${cpreq} ${FIXrrfs}/bumploc/${BUMPLOC}/* bumploc/
 ${cpreq} ${FIXrrfs}/meshes/${MESH_NAME}.static.nc static.nc
-if [[ "${begin}" != "YES" ]]; then
-  cpfs ${DATAROOT}/${NET}/${rrfs_ver}/${RUN}.${CDATEm1:0:8}/${CDATEm1:8:2}/fcst/restart.${timestr}.nc .
-fi
-#${cpreq} ${COMINioda}/..../obs/* obs/                            
-#${cpreq} ${COMINgdas}/..../ens/* ens/
+
+#
+#  link background
+#
+
+ln -snf ${initial_file} .
+
+#
+# copy observations files
+#
+cp ${COMOUT}/ioda_bufr/* obs/.                           
+#
+#  find ensemble forecast
+#
+mpasout_file=mpasout.${timestr}.nc
+for (( ii=0; ii<4; ii=ii+1 )); do
+   CDATEp=$($NDATE -${ii} ${CDATE} )
+   ensdir=${COMINrrfs}/rrfsenkf.${CDATEp:0:8}/${CDATEp:8:2}
+   ensdir_m001=${ensdir}/m001/fcst
+   if [[ -s ${ensdir_m001}/${mpasout_file} ]]; then
+     for (( iii=1; iii<31; iii=iii+1 )); do
+        memid=$(printf %03d ${iii})
+        ln -s ${ensdir}/m${memid}/fcst/${mpasout_file} ens/m${memid}.${mpasout_file}
+     done
```

Comment:
We need to skip the remaining For loop once we find the ensembles, right?

Reply:
I do not have time to get ensemble forecast number and check if ensemble exists. This is hardwired to link 30 members. It good for testing but need more work to make it safe and flexible.

Reply:
I mean that it checks back 4 days, if found on day1, we don't need to check  day2 to day4. This means we can add a `break` line after line 70.